packages/contract-wrappers/src/contract_wrappers/coordinator_wrapper.ts

```diff
@@ -0,0 +1,857 @@
+        assert.doesConformToSchema('orders', orders, schemas.ordersSchema);
+        const makerAddress = getMakerAddressOrThrow(orders);
+        assert.isSenderAddressAsync('makerAddress', makerAddress, this._web3Wrapper);
+        const data = this._transactionEncoder.batchCancelOrdersTx(orders);
+
+        // create lookup tables to match server endpoints to orders
+        const feeRecipientsToOrders: { [feeRecipient: string]: SignedOrder[] } = {};
+        for (const order of orders) {
+            if (feeRecipientsToOrders[order.feeRecipientAddress] === undefined) {
+                feeRecipientsToOrders[order.feeRecipientAddress] = [] as SignedOrder[];
+            }
+            feeRecipientsToOrders[order.feeRecipientAddress].push(order);
+        }
+
+        const serverEndpointsToFeeRecipients: { [feeRecipient: string]: string[] } = {};
+        for (const feeRecipient of Object.keys(feeRecipientsToOrders)) {
+            const endpoint = await this._getServerEndpointOrThrowAsync(feeRecipient);
+            if (serverEndpointsToFeeRecipients[endpoint] === undefined) {
+                serverEndpointsToFeeRecipients[endpoint] = [];
+            }
+            serverEndpointsToFeeRecipients[endpoint].push(feeRecipient);
+        }
+
+        // make server requests
+        let numErrors = 0;
+        let errorResponses: CoordinatorServerResponse[] = [];
+        let successResponses: CoordinatorServerCancellationResponse[] = [];
+        const transaction = await this._generateSignedZeroExTransactionAsync(data, makerAddress);
+        for (const endpoint of Object.keys(serverEndpointsToFeeRecipients)) {
+            const response = await this._executeServerRequestAsync(transaction, makerAddress, endpoint);
+            if (response.isError) {
+                errorResponses = errorResponses.concat(response);
+                numErrors++;
+            } else {
+                successResponses = successResponses.concat(response.body as CoordinatorServerCancellationResponse);
+            }
+        }
+
+        // if no errors
+        if (numErrors === 0) {
+            return successResponses;
+        } else {
+            // lookup orders with errors
+            const errorsWithOrders = errorResponses.map(resp => {
+                const endpoint = resp.coordinatorOperator;
+                const feeRecipients: string[] = serverEndpointsToFeeRecipients[endpoint];
+                const _orders = feeRecipients
+                    .map(feeRecipient => feeRecipientsToOrders[feeRecipient])
+                    .reduce((acc, val) => acc.concat(val));
+                return {
+                    ...resp,
+                    orders: _orders,
+                };
+            });
+
+            // return errors and approvals
+            throw new CoordinatorServerError(
+                CoordinatorServerErrorMsg.CancellationFailed,
+                [],
```

Comment:
For readability, our convention is to declare a variable with a descriptive name rather then passing in values directly. Which is more readable to you:

```
const orders = getOrders(2, [], false);
```

or

```
const limit = 2;
const excludedFields = [];
const isFilled = false;
const orders = getOrders(limit, excludedFields, isFilled);
```

?

Reply:
makes sense! will do